staging/src/k8s.io/kubectl/pkg/cmd/create/create_pdb_test.go

```diff
@@ -17,69 +17,203 @@ limitations under the License.
 package create
 
 import (
-	"bytes"
-	"io/ioutil"
-	"net/http"
 	"testing"
 
-	"k8s.io/apimachinery/pkg/runtime/schema"
-	"k8s.io/cli-runtime/pkg/genericclioptions"
-	restclient "k8s.io/client-go/rest"
-	"k8s.io/client-go/rest/fake"
-	cmdtesting "k8s.io/kubectl/pkg/cmd/testing"
-	"k8s.io/kubectl/pkg/scheme"
+	policyv1beta1 "k8s.io/api/policy/v1beta1"
+	apiequality "k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
 )
 
-func TestCreatePdb(t *testing.T) {
-	pdbName := "my-pdb"
-	tf := cmdtesting.NewTestFactory().WithNamespace("test")
-	defer tf.Cleanup()
-
-	ns := scheme.Codecs.WithoutConversion()
-
-	tf.Client = &fake.RESTClient{
-		GroupVersion:         schema.GroupVersion{Group: "policy", Version: "v1beta1"},
-		NegotiatedSerializer: ns,
-		Client: fake.CreateHTTPClient(func(req *http.Request) (*http.Response, error) {
-			return &http.Response{
-				StatusCode: http.StatusOK,
-				Body:       ioutil.NopCloser(&bytes.Buffer{}),
-			}, nil
-		}),
-	}
-	tf.ClientConfigVal = &restclient.Config{}
+func TestCreatePdbValidation(t *testing.T) {
+	selectorOpts := []string{"app=nginx"}
+	podAmountOpts := []string{"3", "50%"}
 
-	outputFormat := "name"
+	tests := map[string]struct {
+		options  *PodDisruptionBudgetOpts
+		expected string
+	}{
+		"test-missing-name-param": {
+			options: &PodDisruptionBudgetOpts{
+				Selector:     selectorOpts[0],
+				MinAvailable: podAmountOpts[0],
+			},
+			expected: "name must be specified",
+		},
+		"test-missing-selector-param": {
+			options: &PodDisruptionBudgetOpts{
+				Name:         "my-pdb",
+				MinAvailable: podAmountOpts[0],
+			},
+			expected: "a selector must be specified",
+		},
+		"test-missing-max-unavailable-max-unavailable-param": {
+			options: &PodDisruptionBudgetOpts{
+				Name:     "my-pdb",
+				Selector: selectorOpts[0],
+			},
+			expected: "one of min-available or max-unavailable must be specified",
+		},
+		"test--both-min-available-max-unavailable-param": {
+			options: &PodDisruptionBudgetOpts{
+				Name:           "my-pdb",
+				Selector:       selectorOpts[0],
+				MinAvailable:   podAmountOpts[0],
+				MaxUnavailable: podAmountOpts[1],
+			},
+			expected: "min-available and max-unavailable cannot be both specified",
+		},
+		"test-invalid-min-available-format": {
+			options: &PodDisruptionBudgetOpts{
+				Name:         "my-pdb",
+				Selector:     selectorOpts[0],
+				MinAvailable: "10GB",
+			},
+			expected: "invalid format specified for min-available",
+		},
+		"test-invalid-max-unavailable-format": {
+			options: &PodDisruptionBudgetOpts{
+				Name:           "my-pdb",
+				Selector:       selectorOpts[0],
+				MaxUnavailable: "10GB",
+			},
+			expected: "invalid format specified for max-unavailable",
+		},
```

Comment:
Can we also add some tests where the validation passes?

Like one with minAvailable as a percent, other with maxAvailable as a number :)